assets/pages/oneoff-contributions/components/formFields.jsx

```diff
@@ -3,18 +3,111 @@
 // ----- Imports ----- //
 
 import React from 'react';
-import NameFormField from './nameFormField';
-import EmailFormFieldContainer from './emailFormFieldContainer';
+import { connect } from 'react-redux';
+import type { Dispatch } from 'redux';
+import TextInput from 'components/textInput/textInput';
+import {
+  setFullName,
+  setEmail,
+  type Action as UserAction,
+} from 'helpers/user/userActions';
+import { type UserFormFieldAttribute, formFieldIsValid, shouldShowError } from 'helpers/checkoutForm/checkoutForm';
+import {
+  type Action as CheckoutAction,
+  setFullNameShouldValidate,
+  setEmailShouldValidate,
+} from './contributionsCheckoutContainer/checkoutFormActions';
 
 
+// ----- Types ----- //
+
+type PropTypes = {
+  fullName: UserFormFieldAttribute,
+  email: UserFormFieldAttribute,
+  setFullName: (string) => void,
+  setEmail: (string) => void,
+  setFullNameShouldValidate: () => void,
+  setEmailShouldValidate: () => void,
+};
+
+// ----- Map State/Props ----- //
+
+function mapStateToProps(state) {
+
+  const fullName = {
+    value: state.page.user.fullName,
+    ...state.page.checkoutForm.fullName,
+  };
+
+  const email = {
+    value: state.page.user.email,
+    ...state.page.checkoutForm.email,
+  };
+
+  return {
+    fullName: {
+      value: fullName.value,
+      shouldValidate: fullName.shouldValidate,
+      isValid: formFieldIsValid(fullName),
+    },
+    email: {
+      value: email.value,
+      shouldValidate: email.shouldValidate,
+      isValid: formFieldIsValid(email),
+    },
+  };
+
+}
+
+function mapDispatchToProps(dispatch: Dispatch<UserAction | CheckoutAction>) {
+
+  return {
+    setFullNameShouldValidate: () => {
+      dispatch(setFullNameShouldValidate());
+    },
+    setFullName: (fullName: string) => {
+      dispatch(setFullName(fullName));
+    },
+    setEmailShouldValidate: () => {
+      dispatch(setEmailShouldValidate());
+    },
+    setEmail: (email: string) => {
+      dispatch(setEmail(email));
+    },
+  };
+}
```

Comment:
Optional simplification here, I think you could write this as:

```jsx
const mapDispatchToProps = {
  setFullNameShouldValidate,
  setFullName,
  setEmailShouldValidate,
  setEmail,
};
```

I don't think this is necessary, just an FYI.

Reply:
ooh that's nice, I'll take you up on that one!